src/data/list/zip.lean

```diff
@@ -242,4 +264,37 @@ begin
         simp * at *, }, }, },
 end
 
+@[simp] lemma map_uncurry_zip_eq_zip_with
+  (f : α → β → γ) (l : list α) (l' : list β) :
+  map (function.uncurry f) (l.zip l') = zip_with f l l' :=
+begin
+  induction l with hd tl hl generalizing l',
+  { simp },
+  { cases l' with hd' tl',
+    { simp },
+    { simp [hl] } }
+end
+
+@[simp] lemma sum_zip_with_distrib_left [semiring γ]
+  (f : α → β → γ) (n : γ) (l : list α) (l' : list β) :
+  (zip_with (λ x y, n * f x y) l l').sum = n * (l.zip_with f l').sum :=
+begin
+  induction l with hd tl hl generalizing f n l',
+  { simp },
+  { cases l' with hd' tl',
+    { simp, },
+    { simp [hl, mul_add] } }
+end
+
+@[simp] lemma sum_zip_with_distrib_right [semiring γ]
+  (f : α → β → γ) (n : γ) (l : list α) (l' : list β) :
+  (zip_with (λ x y, f x y * n) l l').sum = (l.zip_with f l').sum * n :=
```

Comment:
This lemma is certainly redundant: it follows trivially from the previous lemma and commutativity of multiplication.

Reply:
Removed as the left version is now removed

Reply:
(BTW, I think multiplication was not commutative here, right?!)

Reply:
Ah yes, we only have a `semiring` here. Then the lemmas are actually not as trivial as I thought. Still, I suspect they won't be missed.